src/cx2.cpp

```diff
@@ -59,38 +77,40 @@ struct NNSEMessage {
 	uint16_t    length;		// Length of the packet, including this header
 	uint16_t    seqno;		// Sequence number. Increases by one for every non-ACK packet.
 	uint16_t    csum;		// Checksum. Inverse of the 16bit modular sum with carry added.
+});
 
-	uint8_t     data[0];
```

Comment:
MSVC did *not* like this. I *think* `getPacketData` has the same behavior, though.

Reply:
Nope, `error C2229: struct 'NNSEMessage_AddrReq' has an illegal zero-sized array` either way

Reply:
Try building with `/Za`: https://docs.microsoft.com/en-us/cpp/cpp/arrays-cpp?view=vs-2019

Reply:
> A zero-sized array is legal only when the array is the last field in a struct or union and when the Microsoft extensions are enabled (/Za or /permissive- **isn't** set).

`/Za` *disables* MS extensions, not enables them. And it doesn't allow them in the middle of a struct consecutively (i.e. has a struct that isn't at the end that has one).

Reply:
Right, it has to be "without `/Za`".

https://en.cppreference.com/w/c/language/struct says:

> Structures with flexible array members (or unions who have a recursive-possibly structure member with flexible array member) cannot appear as array elements or as members of other structures.

So this is invalid C99 and arguably should also be rejected as invalid C++ with compiler extensions.
However, this can be worked around by just introducing a struct `NNSEMessageHdr` without the flexible array member and using that as member inside the various request types.

Reply:
right, but that would just be a find/replace for `NNSEMessage` to `NNSEMessageHdr`—I don't think there would be any change.

Reply:
Well, there would be

```
struct NNSEMessage {
	NNSEMessageHdr hdr;
	uint8_t     data[0];
} __attribute__((packed));
```

I tried that and it works, but is actually more messy than just using `getPacketData`. Though the issue with `const` needs to be fixed first.